### PR TITLE
Connect React and Redux

### PR DESCRIPTION
**Pull request by Alejandro-Bernal-M**
# Using redux on the bookstore app

- Add initial state:
https://github.com/Alejandro-Bernal-M/microverse-bookstore/blob/476dbde2ca7c9db43cb8dda4e4310c562338736b/src/redux/books/booksSlice.js#L3-L22

- Displaying the books using redux and adding the delete button:

https://github.com/Alejandro-Bernal-M/microverse-bookstore/blob/476dbde2ca7c9db43cb8dda4e4310c562338736b/src/components/Logic.jsx#L3
 
 https://github.com/Alejandro-Bernal-M/microverse-bookstore/blob/476dbde2ca7c9db43cb8dda4e4310c562338736b/src/components/Logic.jsx#L6

https://github.com/Alejandro-Bernal-M/microverse-bookstore/blob/476dbde2ca7c9db43cb8dda4e4310c562338736b/src/components/Logic.jsx#L19-L40

- Add the add new book functionality using redux: 

https://github.com/Alejandro-Bernal-M/microverse-bookstore/blob/476dbde2ca7c9db43cb8dda4e4310c562338736b/src/components/BooksForm.jsx#L4

https://github.com/Alejandro-Bernal-M/microverse-bookstore/blob/476dbde2ca7c9db43cb8dda4e4310c562338736b/src/components/BooksForm.jsx#L14-L16

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,22 +1,40 @@
...
+    author: 'Leo Tolstoy',
+    category: 'Fiction',
+  },
+  {
+    item_id: 'item3',
+    title: 'The Selfish Gene',
+    author: 'Richard Dawkins',
+    category: 'Nonfiction',
+  },
+];
```
Comment (GzimAsani) [inline src/redux/books/booksSlice.js]: - [ ] Everything works fine with regards to books :clap: But for categories: when clicking the button the message should show, kindly fix it

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @Alejandro-Bernal-M                     :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with the usage of useSeelector and dispatch  :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (Alejandro-Bernal-M) [reply src/redux/books/booksSlice.js]: Hello @GzimAsani would you please show me in what part of the project [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/use_redux_in_react_components_v2.md) says that part about the categories?
 ## Project requirements
 - Wrap your <App> with the <Provider> component from react-redux
- Pass your configured store into the <Provider>
- Create an initialState variable for your booksSlice, which will include the following:
`
// Initial state:
[
  {
    "item_id": "item1",
    "title": "The Great Gatsby",
    "author": "John Smith",
    "category": "Fiction"
  },
  {
    "item_id": "item2",
    "title": "Anna Karenina",
    "author": "Leo Tolstoy",
    "category": "Fiction"
  },
  {
    "item_id": "item3",
    "title": "The Selfish Gene",
    "author": "Richard Dawkins",
    "category": "Nonfiction"
  }
]
`
- Display your books, received from the slice, in a reusable component
- Dispatch actions using useDispatch
- Add a <Button> component, which includes:
- An event handler that adds a book to the books array (with attributes id, title and author)
- Add a <Button> component, which includes:
- An event handler that removes a book from the books array (by id)

**Review by nicupop729 — COMMENTED** [review]
Hello @Alejandro-Bernal-M  👋,

_This is an additional review as a result of the second opinion request._

Kudos for the great work done on this project! :clap:

I can also confirm that the app is running as expected and that the project requirements are being followed.

As it was not mentioned in the project requirements that the "Check status" button should have a function, you can disregard the comment from the reviewer.

Usually, because you were right to ask for a second opinion and because the reviewer should not have asked about aspects not mentioned in the project requirements, the Code Review Request should be approved right away by the next reviewer. But because you introduced new changes to the project, the next reviewer will provide a thorough check of the new changes, and only if everything is in order will the project be approved.

Next time, after you receive a review you don't agree with and after you ask for a Second Opinion review, please don't push other commits before the SO is provided.

Please ask for another review in your Dashboard:)

Keep rocking 🚀!

**Comment by Alejandro-Bernal-M** [comment]
@nicupop729 Ok, I understand, thank you. Have a nice day. 😄

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @Alejandro-Bernal-M 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Linters checks are passing
- ✔️ Use Git Flow
- ✔️ Meaningful commit messages.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**